Source/Core/Editor/UI/Drawers/ProcessSceneReferenceDrawer.cs

```diff
@@ -453,11 +454,27 @@ protected void SceneObjectAutomaticSetup(GameObject selectedSceneObject, Type va
         {
             ISceneObject sceneObject = selectedSceneObject.GetComponent<ProcessSceneObject>() ?? selectedSceneObject.AddComponent<ProcessSceneObject>();
 
-            if (typeof(ISceneObjectProperty).IsAssignableFrom(valueType))
-            {
-                sceneObject.AddProcessProperty(valueType);
+            // Find the first public, non-abstract class implementing valueType and having valueType as DefaultImplementationAttribute
+            Type concreteTypeToAdd = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
There's the utility method ReflectionUtils.GetConcreteImplementationsOf you could use to shorten this. There might be something about attributes too, or it might be worth adding these functionalities to the ReflectionUtils class.